File: springboot-kotlin-doma/src/main/java/com/example/demo/repository/LanguageRepository.java
package com.example.demo.repository;

import java.util.List;

import com.example.demo.entity.Language;
import org.seasar.doma.Dao;
import org.seasar.doma.Select;
import org.seasar.doma.boot.ConfigAutowireable;

@ConfigAutowireable
@Dao
public interface LanguageRepository {
    @Select
    List<Language> findAll();
}

File: springboot-kotlin-doma/src/main/kotlin/com/example/demo/repository/LanguageRepository.kt
package com.example.demo.repository

import com.example.demo.entity.LanguageEntity
import org.seasar.doma.Dao
import org.seasar.doma.Select
import org.seasar.doma.boot.ConfigAutowireable

@ConfigAutowireable
@Dao
interface LanguageRepository {
    @Select
    fun findAll(): List<LanguageEntity>
}
